Classes/Controller/TimelineController.php

```diff
@@ -105,18 +107,55 @@ public function timelineAction()
     {
         $zoom = 15; // amount of seconds per pixel
 
-        $groupedLogs = $this->logRepository->findGroupedByTask();
+        $startTime = strtoTime('-1 hour');
+        $endTime = time();
+        $postVars = GeneralUtility::_POST('tx_schedulertimeline_tools_schedulertimelineschedulertimeline');
+        $timespan = $postVars['timespan'] ?? null;
+        if ($timespan !== null) {
+            $startTime = $this->getStartTime($postVars['timespan']);
+            $endTime = time();
+        }
+        $groupedLogs = $this->logRepository->findGroupedByTask($startTime, $endTime);
 
-        $starttime = $this->hourFloor($this->logRepository->getMinDate());
-        $endtime = $this->hourCeil($this->logRepository->getMaxDate());
+        // To have logs that was added this second included,
+        $startTimeFloor = $this->hourFloor($startTime);
+        $endTimeCeil = $this->hourCeil($endTime);
 
         $this->view->assign('groupedLogs', $groupedLogs);
-
-        $this->view->assign('starttime', $starttime);
-        $this->view->assign('endtime', $endtime);
+        $this->view->assign('starttime', $startTimeFloor);
+        $this->view->assign('endtime', $endTimeCeil);
         $this->view->assign('zoom', $zoom);
-        $this->view->assign('now', ($GLOBALS['EXEC_TIME'] - $starttime) / $zoom);
-        $this->view->assign('timelinePanelWidth', ($endtime - $starttime) / $zoom);
+        $this->view->assign('now', ($GLOBALS['EXEC_TIME'] - $startTimeFloor) / $zoom);
+        $this->view->assign('timelinePanelWidth', ($endTimeCeil - $startTimeFloor) / $zoom);
+        $this->view->assign('actionMenuValues', $this->getActionMenuValues());
+        $this->view->assign('actionMenuSelected', $timespan);
+    }
+
+    /**
+     * @param string $timespan
+     * @return int
+     */
+    private function getStartTime(string $timespan)
+    {
+        $strToTimeValue = FormValueTimestampConverter::convertValueToStrToTimeValue($timespan);
+        return strtotime('-' . $strToTimeValue);
```

Comment:
Idea: move the `strtotime` call into the `convertValueToStrToTimeValue` and make it a `convertFormValueToTimestamp` to remove the `getStartTime` function.
As the `getStartTime` function is only called once, in direct relation to `convertValueToStrToTimeValue`.